library/src/main/java/kwasm/format/text/Type.kt

```diff
@@ -14,37 +14,54 @@
 
 package kwasm.format.text
 
+import kwasm.ast.ValueType
 import kwasm.format.ParseContext
+import kwasm.format.ParseException
 
-sealed class Type(
+sealed class Type<T>(
     protected val sequence: CharSequence,
     protected val context: ParseContext? = null
 ) {
+    val value: T by lazy {
+        parseValue()
+    }
```

Comment:
nit: this could be one-lined:  `val value: T by lazy { parseValue() }`

Reply:
Done